src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/BeneficialOwnerAdditionService.java

```diff
@@ -0,0 +1,180 @@
+package uk.gov.companieshouse.overseasentitiesapi.service;
+
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.AddressDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.BeneficialOwnerCorporateDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.BeneficialOwnerGovernmentOrPublicAuthorityDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.BeneficialOwnerIndividualDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.OverseasEntitySubmissionDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.PersonName;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.additions.Addition;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.additions.CorporateEntityBeneficialOwnerAddition;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.additions.IndividualBeneficialOwnerAddition;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.additions.LegalPersonBeneficialOwnerAddition;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.commonmodels.Address;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static uk.gov.companieshouse.overseasentitiesapi.utils.NatureOfControlTypeMapping.collectAllNatureOfControlsIntoSingleList;
+
+@Service
+public class BeneficialOwnerAdditionService {
+    public List<Addition> beneficialOwnerAdditions(OverseasEntitySubmissionDto overseasEntitySubmissionDto) {
+        List<Addition> additions = new ArrayList<>();
+
+        additions.addAll(getIndividualBeneficialOwnerAdditions(overseasEntitySubmissionDto)
+                .stream()
+                .flatMap(Optional::stream)
+                .collect(Collectors.toList()));
+
+        additions.addAll(getCorporateEntityBeneficialOwnerAdditions(overseasEntitySubmissionDto)
+                .stream()
+                .flatMap(Optional::stream)
+                .collect(Collectors.toList()));
+
+        additions.addAll(getLegalPersonBeneficialOwnerAdditions(overseasEntitySubmissionDto)
+                .stream()
+                .flatMap(Optional::stream)
+                .collect(Collectors.toList()));
+
+        return additions;
+    }
+
+    private List<Optional<IndividualBeneficialOwnerAddition>> getIndividualBeneficialOwnerAdditions(
+            OverseasEntitySubmissionDto overseasEntitySubmissionDto) {
+        var beneficialOwnersIndividual = overseasEntitySubmissionDto.getBeneficialOwnersIndividual();
+        return beneficialOwnersIndividual.stream()
+                .filter(beneficialOwner -> (beneficialOwner.getChipsReference() == null))
+                .map(this::getIndividualBeneficialOwnerAddition)
+                .collect(Collectors.toList());
```

Comment:
is collect needed?

Reply:
With Java Streams you need a terminal function to obtain the result of the Stream 

Without `.collect()` this would just continue being a stream

Reply:
Sure at some stage you need collect() but you could return a stream from these private methods.

Reply:
I'm fine with collecting intermediates into Lists but it's really just:
source.stream().filter().map().forEachOrdered(target::add);

Reply:
I've implemented this by passing in the top  'additions' into each method and changing it to use the forEachOrdered function:
`beneficialOwnersIndividual.stream()
            .filter(beneficialOwner -> (beneficialOwner.getChipsReference() == null))
            .map(this::getIndividualBeneficialOwnerAddition)
            .forEachOrdered(additions::add);`